pkg/occlient/occlient_test.go

```diff
@@ -529,3 +615,576 @@ func Test_parseImageName(t *testing.T) {
 		})
 	}
 }
+
+func TestNewAppS2I(t *testing.T) {
+	type args struct {
+		name         string
+		builderImage string
+		gitUrl       string
+		labels       map[string]string
+		annotations  map[string]string
+	}
+
+	tests := []struct {
+		name      string
+		namespace string
+		args      args
+		wantErr   bool
+	}{
+		{
+			name: "with valid gitUrl",
+			args: args{
+				name:         "ruby",
+				builderImage: "ruby:latest",
+				gitUrl:       "https://github.com/openshift/ruby",
+				labels: map[string]string{
+					"app": "apptmp",
+					"app.kubernetes.io/component-name": "ruby",
+					"app.kubernetes.io/component-type": "ruby",
+					"app.kubernetes.io/name":           "apptmp",
+				},
+				annotations: map[string]string{
+					"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+					"app.kubernetes.io/component-source-type": "git",
+				},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+
+			fkclientset.ImageClientset.PrependReactor("list", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStreams(), nil
+			})
+
+			fkclientset.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStream(), nil
+			})
+
+			fkclientset.ImageClientset.PrependReactor("get", "imagestreamimages", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStreamImages(), nil
+			})
+
+			err := fkclient.NewAppS2I(tt.args.name,
+				tt.args.builderImage,
+				tt.args.gitUrl,
+				tt.args.labels,
+				tt.args.annotations)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewAppS2I() error = %#v, wantErr %#v", err, tt.wantErr)
+			}
+
+			if len(fkclientset.ImageClientset.Actions()) != 3 {
+				t.Errorf("expected 3 ImageClientset.Actions() in NewAppS2I, got: %v", fkclientset.ImageClientset.Actions())
+			}
+
+			if len(fkclientset.BuildClientset.Actions()) != 1 {
+				t.Errorf("expected 1 BuildClientset.Actions() in NewAppS2I, got: %v", fkclientset.BuildClientset.Actions())
+			}
+
+			if len(fkclientset.AppsClientset.Actions()) != 1 {
+				t.Errorf("expected 1 AppsClientset.Actions() in NewAppS2I, go: %v", fkclientset.AppsClientset.Actions())
+			}
+
+			if len(fkclientset.Kubernetes.Actions()) != 1 {
+				t.Errorf("expected 1 Kubernetes.Actions() in NewAppS2I, go: %v", fkclientset.Kubernetes.Actions())
+			}
+
+			if err == nil {
+				createdIS := fkclientset.ImageClientset.Actions()[2].(ktesting.CreateAction).GetObject().(*imagev1.ImageStream)
```

Comment:
why are we checking just `3rd` action?  What if the order in which ImageStream is created changes?